OSX add option to force dedicated GPU

Akemi opened the pull request:
i can't test it myself and i also don't feel very strongly about it. it fixes #3242 and possibly partially #3653. it could also be used as a workaround for #2371.

i dropped support for OS X 10.7 and earlier to simplify the context creation which is a bit buggy because of the if directive. in some circumstances it could remove kCGLPFAAccelerated instead of kCGLPFASupportsAutomaticGraphicsSwitching, which would only be fixable with another if directive. 10.7 isn't supported officially anymore anyway

wm4 commented:
Seems ok. Maybe the option should be excluded on non-OSX, since it would be too misleading on the other platforms. Might be worth thinking about declaring the option inside the cocoa backend.

Akemi commented:
is this what you had in mind? 
renamed cgl_context to priv since it now has the opts struct inside and is not exclusive to the cgl context anymore.
moved the description from the **Video** to **OpenGL renderer options** category in the manual right underneath the Angle specific ones. i think like this it makes more sense?

wm4 commented:
Yeah, good work.
